uploadNeuroDB/NeuroDB/MRIProcessingUtility.pm

```diff
@@ -1770,72 +1785,84 @@ or a phantom
-	if ($this->{verbose});
+
     if ($sth->rows == 0) {
         print LOG  "\n\n=> Could not find candidate with CandID =".
                    " $subjectIDsref->{'CandID'} in database\n";
         $CandMismatchError = 'CandID does not exist';
+
         return $CandMismatchError;
     }
-    
+
+    print "candidate id $subjectIDsref->{'CandID'}\n" if ($this->{verbose});
+
+
     #################################################################
     ### Check if PSCID exists and that PSCID and CandID of the scan
     # refers to the same candidate
     #################################################################
     $query = "SELECT CandID, PSCID FROM candidate WHERE PSCID=?";
-    $sth =  ${$this->{'dbhr'}}->prepare($query);
+    $sth   = ${$this->{'dbhr'}}->prepare($query);
     $sth->execute($subjectIDsref->{'PSCID'});
+
     if ($sth->rows == 0) {
         print LOG "\n=> No PSCID\n";
         $CandMismatchError= 'PSCID does not exist';
+
         return $CandMismatchError;
     } else {
         # Check that the PSCID and CandID refers to the same candidate
         my $rowref = $sth->fetchrow_hashref;
         unless ($rowref->{'CandID'} == $subjectIDsref->{'CandID'}) {
             $CandMismatchError = 'PSCID and CandID of the image mismatch\n';
             print LOG "\n=> $CandMismatchError";
+
             return $CandMismatchError;
         }
     }
 
-    ############################################################
-    ################ No Checking if the subject is Phantom #####
-    ############################################################
-    if ($subjectIDsref->{'isPhantom'}) {
-        # CandID/PSCID errors don't apply to phantoms, so we don't
-        # want to trigger
-        # the check which aborts the insertion
-        $CandMismatchError = undef;
-        return $CandMismatchError;
-    }
 
     ############################################################
-    ################ Check if visitLabel exists ################
+    ################ Check if visit label is valid##############
     ############################################################
+    my $visit_label = $subjectIDsref->{'visitLabel'};
 
     $query = "SELECT Visit_label FROM Visit_Windows WHERE BINARY Visit_label=?";
-    $sth =  ${$this->{'dbhr'}}->prepare($query);
-    $sth->execute($subjectIDsref->{'visitLabel'});
+    $sth   = ${$this->{'dbhr'}}->prepare($query);
+    $sth->execute($visit_label);
+
     if (($sth->rows == 0) && (!$subjectIDsref->{'createVisitLabel'})) {
         print LOG "\n=> No Visit label\n";
-        $CandMismatchError= 'Visit label does not exist';
+        $CandMismatchError = "Visit label $visit_label does not exist in Visit_Windows";
+
         return $CandMismatchError;
     } elsif (($sth->rows == 0) && ($subjectIDsref->{'createVisitLabel'})) {
-        print LOG "\n=> Will create visit label $subjectIDsref->{'visitLabel'}\n";
+        print LOG "\n=> Will create visit label $visit_label in Visit_Windows\n";
```

Comment:
this is a more appropriate log error as it always get confusing as to why the visit label is deemed invalid. It is invalid only if it is not part of the Visit_Windows table.